### PR TITLE
OCP-2316 OCP-2351

### PR DESCRIPTION
**Pull request by mdimopoulos**
AAC conformance test suites + necessary changes and tiny refactoring.

I don't understand why it has 3 commits, it should be 2. I rebased from master 4 times.
I also see changes that i had done in OCP-2317 appear here again, although that branch was merged to master :S.

### CODE REVIEW

**Comment by rgonzalezfluendo** [comment]
Strange. I got conflicts rebasing with master. (There are easy to handle).

```
$ git fetch
$ git rebase orrigin/master
$ git diff
diff --cc fluster/decoders/fdk_aac.py
index 367972a,ee63726..0000000
--- fluster/decoders/fdk_aac.py
+++ fluster/decoders/fdk_aac.py
@@@ -19,7 -19,7 +19,11 @@@
  # Free Software Foundation, Inc., 59 Temple Place - Suite 330,
  # Boston, MA 02111-1307, USA.

++<<<<<<< HEAD
 +from fluster.codec import Codec, PixelFormat
++=======
+ from fluster.codec import Codec
++>>>>>>> Add AAC reference decoder.
  from fluster.decoder import Decoder, register_decoder
  from fluster.utils import file_checksum, run_command

@@@ -32,7 -32,7 +36,11 @@@ class FDKAACDecoder(Decoder)
      codec = Codec.AAC
      binary = 'aac-dec'

++<<<<<<< HEAD
 +    def decode(self, input_filepath: str, output_filepath: str, output_format: PixelFormat, timeout: int,
++=======
+     def decode(self, input_filepath: str, output_filepath: str, output_format: None, timeout: int,
++>>>>>>> Add AAC reference decoder.
                 verbose: bool) -> str:
          '''Decodes input_filepath in output_filepath'''
          run_command([self.binary, input_filepath, output_filepath],
```

Let me know if you need help.

**Comment by mdimopoulos** [comment]
Ok i figured it out, i pulled the latest from remote master.
I'll cancel this pull request and create a new one.

**Comment by pamarcos** [comment]
@mdimopoulos 

> Ok i figured it out, i pulled the latest from remote master.

There's no need to create a new PR. You can use this one as long as the branch is the same. You can force-push the changes onto this branch and good to go.

File: fluster/test.py [hunk 1]
```diff
@@ -56,7 +56,15 @@ def __init__(
...
-        output_filepath = os.path.join(self.results_dir, self.test_vector.name + ".yuv")
+        if self.decoder.name == "FDK-AAC":
+            output_filepath = os.path.join(
+                self.results_dir, self.test_vector.name + ".wav"
+            )
+        else:
+            output_filepath = os.path.join(
+                self.results_dir, self.test_vector.name + ".yuv"
+            )
+
```
Comment (pamarcos) [inline fluster/test.py:67]: All this should be on the other commit: https://github.com/fluendo/fluster/pull/87/commits/ef10c6004bc36684afb3fbd82ce6bd8213ac65b3

Comment (rgonzalezfluendo) [reply fluster/test.py:67]: Do we need the extension? no IMHO. At the end, the output_filepath is only a temporal file. We can use for yuv and for wav (and for rbga):
```
output_filepath = os.path.join(self.results_dir, self.test_vector.name + 'raw')
```

**Review by pamarcos — APPROVED** [review]
LGTM! But of course, I did half the commits, so we'd better wait for @rgonzalezfluendo's review :)

File: fluster/decoders/gstreamer.py [hunk 2]
```diff
@@ -43,6 +43,14 @@ def gst_element_exists(element: str) -> bool:
     return False
 
 
+def output_format_to_gst(output_format: OutputFormat) -> str:
+    """Return GStreamer pixel format"""
+    mapping = {OutputFormat.YUV420P: "I420", OutputFormat.YUV420P10LE: "I420_10LE"}
+    if not mapping.get(output_format):
```
Comment (rgonzalezfluendo) [inline fluster/decoders/gstreamer.py:49]:
```suggestion
    if output_format not in mapping:
```

**Review by rgonzalezfluendo — APPROVED** [review]

File: test_suites/h264/JVT-AVC_V1.json [hunk 3]
```diff
@@ -1017,15 +1017,15 @@
...
+            "result": "c086a2dd4a002734ee4d385a797a3536"
         },
         {
             "name": "sp2_bt_b",
             "source": "https://www.itu.int/wftp3/av-arch/jvt-site/draft_conformance/AVCv1/sp2_bt_b.zip",
             "source_checksum": "c1601a80dabffd65a1c3416635b6e862",
             "input_file": "H26L/BitstreamExchange/sp2_bt_b.h264",
             "output_format": "yuv420p",
-            "result": "77fef826070d3d17174529b16765bb24"
+            "result": "6de46a828ea40399b0741efd0bdfea11"
```
Comment (pamarcos) [inline test_suites/h264/JVT-AVC_V1.json:1028]: The reference decoder does not work correctly for these two, apparently. We knew about this. We took the results from the .zip themselves.

File: fluster/test_suite.py [hunk 4]
```diff
@@ -266,35 +282,37 @@ def _run_worker(self, test: Test) -> TestVector:
...
+                        f"[{test_result.name}]\t({decoder.name})\t{test_result.name:{max_len}} ... \
+                        {test_result.reference_result.value}",
+                        flush=True,
+                    )
+                else:
+                    print(
+                        f"[{test_result.name}]\t({decoder.name})\t{test_result.name:{max_len}} ... \
+                        {test_result.test_result.value}",
+                        flush=True,
+                    )
```
Comment (rgonzalezfluendo) [inline fluster/test_suite.py:315]:
```suggestion
                value = test_result.test_result.value if reference else test_result.reference_result.value
                print(
                    f"[{test_result.name}]\t({decoder.name})\t{test_result.name:{max_len}} ... \
                    {value}",
                    flush=True,
                )
```

**Pull request merged**